src/ui/server/controllers/insurance/declarations/map-submitted-data/modern-slavery/index.ts

```diff
@@ -27,6 +36,22 @@ const mapSubmittedData = (formBody: RequestBody): object => {
     populatedData[IS_NOT_AWARE_OF_EXISTING_SLAVERY] = null;
   }
 
+  /**
+   * If any of the following fields are true,
+   * The related conditional fields (for an answer of false) should be wiped.
+   */
+  if (formBody[WILL_ADHERE_TO_ALL_REQUIREMENTS] === 'true') {
```

Comment:
Can this please be `switch-case` for multiple conditions.

Reply:
Good shout - Tried a few things and I do not think we should use a switch/case. E.g, this will not work because of TypeScript (`Type 'boolean' is not comparable to type 'RequestBody'`)

```ts
switch (formBody) {
  case isEmptyString(formBody[WILL_ADHERE_TO_ALL_REQUIREMENTS]):
    populatedData[WILL_ADHERE_TO_ALL_REQUIREMENTS] = null;

  case formBody[WILL_ADHERE_TO_ALL_REQUIREMENTS] === '':
    populatedData[WILL_ADHERE_TO_ALL_REQUIREMENTS] = null;

  case formBody[WILL_ADHERE_TO_ALL_REQUIREMENTS] === 'true':
    populatedData[CANNOT_ADHERE_TO_ALL_REQUIREMENTS] = '';
```

The only alternative I can think of would be to have multiple switch/cases for individual fields which I think is uglier, and harder to read/maintain

```ts
  switch (formBody[WILL_ADHERE_TO_ALL_REQUIREMENTS]) {
    case isEmptyString(formBody[WILL_ADHERE_TO_ALL_REQUIREMENTS]):
      populatedData[WILL_ADHERE_TO_ALL_REQUIREMENTS] = null;

    case formBody[WILL_ADHERE_TO_ALL_REQUIREMENTS] === 'true':
      populatedData[CANNOT_ADHERE_TO_ALL_REQUIREMENTS] = '';
  }

  switch (formBody[HAS_NO_OFFENSES_OR_INVESTIGATIONS]) {
    case isEmptyString(formBody[HAS_NO_OFFENSES_OR_INVESTIGATIONS]):
      populatedData[HAS_NO_OFFENSES_OR_INVESTIGATIONS] = null;

    case formBody[HAS_NO_OFFENSES_OR_INVESTIGATIONS] === 'true':
      populatedData[OFFENSES_OR_INVESTIGATIONS] = '';
  }

  switch (formBody[IS_NOT_AWARE_OF_EXISTING_SLAVERY]) {
    case isEmptyString(formBody[IS_NOT_AWARE_OF_EXISTING_SLAVERY]):
      populatedData[IS_NOT_AWARE_OF_EXISTING_SLAVERY] = null;

    case formBody[IS_NOT_AWARE_OF_EXISTING_SLAVERY] === 'true':
      populatedData[AWARE_OF_EXISTING_SLAVERY] = '';
  }
```

I think the existing if statements is much cleaner. Thoughts?